File: Competitive Programming 3/Section 1.4/Game (Card)/10646 - What is the Card?/C++/main.cpp
#include <bits/stdc++.h>

using namespace std;

int getValue(const string & card){
    if(isdigit(card[0]))
        return card[0]-'0';
    return 10;
}
int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int t;
    string card;
    cin >> t ; 
    for(int cs = 1 ; cs <= t ; cs++){
        vector<string>st;
        queue<string>temp;
        for(int i = 0 ; i < 27 ; i ++){
            cin >> card;
            st.push_back(card);
        }
        for(int i = 0 ; i < 25; i ++){
            cin >> card;
            temp.push(card);
        }
        int y=0;
        for(int i = 0 ; i < 3 ; i ++){
            int val = getValue(st.back());
            st.pop_back();
            y+=val;
            for(int j = 10 - val ; j > 0; j --)
                st.pop_back();
        }
        while(!temp.empty()){
            st.push_back(temp.front());
            temp.pop();
        }
        cout << "Case " << cs << ": " << st[y-1] << "\n";
    }
    return 0;
}

File: Competitive Programming 3/Section 1.4/Game (Card)/10646 - What is the Card?/Python/main.py
from queue import Queue

def getValue(card):
    if str.isdigit(card[0]):
        return ord(card[0])-ord('0')
    return 10
def main():
    t = int(input())+1
    for cs in range(1,t):
        st,temp=[],Queue()
        i = 0
        while i < 52:
            for tk in input().split():
                if i < 27:
                    st.append(tk)
                else:
                    temp.put(tk)
                i+=1
        y=0
        for _ in range(3):
            val = getValue(st.pop())
            y+=val
            for _ in range(0,10-val):
                st.pop()
        while not temp.empty():
            st.append(temp.get())
        print("Case "+str(cs)+": "+st[y-1])
main()